apps/avifdec.c

```diff
@@ -337,78 +333,139 @@ int main(int argc, char * argv[])
+        printf("   * Decoded frame [%d] [pts %2.2f (%" PRIu64 " timescales)] [duration %2.2f (%" PRIu64 " timescales)] [%ux%u]\n",
+               currIndex,
+               decoder->imageTiming.pts,
+               decoder->imageTiming.ptsInTimescales,
+               decoder->imageTiming.duration,
+               decoder->imageTiming.durationInTimescales,
+               decoder->image->width,
+               decoder->image->height);
+        if (infoOnly) {
+            ++currIndex;
+            if (decodeAllFrames) {
+                continue;
+            } else {
+                break;
+            }
         }
-        if (!y4mWrite(outputFilename, decoder->image)) {
-            goto cleanup;
+
+        if (decoder->image->transformFlags & AVIF_TRANSFORM_CLAP) {
+            avifCropRect cropRect;
+            if (!avifCropRectFromCleanApertureBox(&cropRect,
+                                                  &decoder->image->clap,
+                                                  decoder->image->width,
+                                                  decoder->image->height,
+                                                  &decoder->diag)) {
+                // Should happen only if AVIF_STRICT_CLAP_VALID is disabled.
+                fprintf(stderr, "Warning: Invalid Clean Aperture values\n");
+            }
         }
-    } else if (outputFormat == AVIF_APP_FILE_FORMAT_JPEG) {
-        // Bypass alpha multiply step during conversion
-        if (rawColor) {
-            decoder->image->alphaPremultiplied = AVIF_TRUE;
+
+        if (ignoreICC && (decoder->image->icc.size > 0)) {
+            printf("[--ignore-icc] Discarding ICC profile.\n");
+            // This cannot fail.
+            result = avifImageSetProfileICC(decoder->image, NULL, 0);
+            assert(result == AVIF_RESULT_OK);
         }
-        if (!avifJPEGWrite(outputFilename, decoder->image, jpegQuality, chromaUpsampling)) {
-            goto cleanup;
+
+        if (iccOverrideFilename) {
+            printf("[--icc] Setting ICC profile: %s\n", iccOverrideFilename);
+            result = avifImageSetProfileICC(decoder->image, iccOverride.data, iccOverride.size);
+            if (result != AVIF_RESULT_OK) {
+                fprintf(stderr, "ERROR: Failed to set ICC: %s\n", avifResultToString(result));
+                goto cleanup;
+            }
         }
-    } else if (outputFormat == AVIF_APP_FILE_FORMAT_PNG) {
-        if (!avifPNGWrite(outputFilename, decoder->image, requestedDepth, chromaUpsampling, pngCompressionLevel)) {
+
+        if (decodeAllFrames) {
+            // Create filename for individual frames, in the form path/to/output-000.ext
+            char * lastDot = strrchr(outputFilename, '.');
+            const size_t dotPos = (lastDot != NULL) ? (size_t)(lastDot - outputFilename) : strlen(outputFilename);
+            const char * extension = (lastDot != NULL) ? lastDot + 1 : "";
```

Comment:
Not sure if a feature or not but passing a folder will have different behaviors depending on whether it ends with a `/` or not:
- `some/output` will create files `some/output0000000000`
- `some/output/` will create files `some/output/0000000000`

This may be tricky but fine. It is a shame that this is not C++ where we could leverage `std::filesystem` for all this.

Reply:
You can't pass in a directory because avifdec wouldn't be able to determine the output file type. In both of your examples it will fail with 'Cannot determine output file extension'

Reply:
I thought there was some `avifdec --output_format png` but I was mistaken.